File: Simplify Path.py
class Solution:
	# @param path, a string
	# @return a string
	def simplifyPath(self, path):
		dirs = path.split("/")
		result = []
		for dir in dirs:
			if (dir == "." or dir == ""):
				continue
			elif (dir == ".."):
				if (result):
					result.pop()
			else:
				result.append(dir)
		str = ""
		for dir in result:
			str += "/" + dir
		if (str == ""):
			return "/"
		return str

File: Simplify Path.java
public class Solution {
    public String simplifyPath(String path) {
        String[] dirs = path.split("/");
        Deque<String> deq = new ArrayDeque<String>();
        for (int i = 0; i < dirs.length; i++) {
            if (dirs[i].equals("")) {
                continue;
            }
            else if (dirs[i].equals(".")) {
                continue;
            }
            else if (dirs[i].equals("..")) {
                if (!deq.isEmpty())
                    deq.removeLast();
            }
            else 
            deq.addLast(dirs[i]);
        }
        
        StringBuilder result = new StringBuilder();
        while (!deq.isEmpty()) {
            result.append("/");
            String dir = deq.removeFirst();
            result.append(dir);
        }
        if (result.toString().equals(""))
            result.append("/");
        return result.toString();
    }
}